packages/angular-output-target/src/generate-proxies.ts

```diff
@@ -70,19 +70,23 @@ function getProxy(cmpMeta: ComponentCompilerMeta) {
   if (inputs.length > 0) {
     directiveOpts.push(`inputs: ['${inputs.join(`', '`)}']`);
   }
+  if (outputs.length > 0) {
+    directiveOpts.push(`outputs: ['${outputs.join(`', '`)}']`)
+  }
 
   const tagNameAsPascal = dashToPascalCase(cmpMeta.tagName);
-  const lines = [`
+
+  const outputsInterface =
+  outputs.length > 0
+    ? `export declare interface ${tagNameAsPascal} extends Required<Pick<JSX.${tagNameAsPascal}, '${outputs.map(o => 'on' + o.replace(/^./, match => match.toUpperCase())).join(`' | '`).replace(/\| $/g, '')}'>> {}`
```

Comment:
Here, the only available import I could find is `JSX.tagNameAsPascal`. Ideally this should come from `dist/types/components/component/component.d.ts`. That way, we can get rid of adding the `on` prefix and correctly inherit the type EventEmitter<WhateverType>.

Pick - will take only the events mentioned in the union
Required - will mark the selected events as required to match the previous implementation.

Reply:
I can see how custom names for events would be a problem so I moved from this strategy to the following:

```diff
import { Components } from 'component-library';
import { Button as IButton } from 'component-library/dist/types/components/my-button/my-button'
-- export declare interface MyButton extends Required<Pick<IButton, 'ionFocus'>> {}
export declare interface MyButton extends Components.MyButton {}
@ProxyCmp({ inputs: ["buttonType"] })
@Component({
  selector: "my-button",
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: "<ng-content></ng-content>",
  inputs: ["buttonType"],
  outputs: ["ionFocus"]
})
export class MyButton {
++  /** Emitted after the popover has presented. */
++  ionButtonFocus!: IButton['ionFocus'];
  constructor(c: ChangeDetectorRef, r: ElementRef, protected z: NgZone) {
    c.detach();
    this.el = r.nativeElement;
    proxyOutputs(this, this.el, ["ionFocus"]);
  }
}
```